src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/HotNodeRca.java

```diff
@@ -94,7 +94,7 @@ public ResourceFlowUnit operate() {
       // reset the variables
       counter = 0;
       hasUnhealthyFlowUnit = false;
-      return new ResourceFlowUnit(System.currentTimeMillis(), context, summary);
+      return new ResourceFlowUnit(System.currentTimeMillis(), context, summary, !currentNode.getIsMasterNode());
```

Comment:
for readability, we might want to add this to a variable with appropriate name and add a line of comment why the masterNode does not need to persist the summary.